core/domain/user_jobs_one_off_test.py

```diff
@@ -589,6 +598,58 @@ def test_weekly_stats_for_multiple_explorations(self):
             }
         }])
 
+    def test_stats_for_multiple_weeks(self):
+        exploration = self.save_new_valid_exploration(
+            self.EXP_ID_1, self.owner_id)
+        exp_id = exploration.id
+        init_state_name = exploration.init_state_name
+        self._rate_exploration('user1', exp_id, 4)
+        self._record_play(exp_id, init_state_name)
+        self._record_play(exp_id, init_state_name)
+
+        (user_jobs_continuous_test.ModifiedUserStatsAggregator.
+         start_computation())
+        self.process_and_flush_pending_tasks()
+
+        self._run_one_off_job()
+        weekly_stats = user_services.get_weekly_dashboard_stats(self.owner_id)
+        self.assertEqual(weekly_stats, [{
+            utils.get_current_date_as_string(): {
+                'average_ratings': 4.0,
+                'total_plays': 2
+            }
+        }])
+
+        (user_jobs_continuous_test.ModifiedUserStatsAggregator.
+         stop_computation(self.owner_id))
+        self.process_and_flush_pending_tasks()
+
+        self._rate_exploration('user2', exp_id, 2)
+
+        (user_jobs_continuous_test.ModifiedUserStatsAggregator.
+         start_computation())
+        self.process_and_flush_pending_tasks()
+
+        def _mock_get_date_after_one_week():
+            """Returns the date of the next week."""
+            return (
```

Comment:
Fragile. Compute the date once (outside this function) and return a constant here, otherwise each time you call this function you get a slightly different number.